homeassistant/components/dhcp/__init__.py

```diff
@@ -32,72 +46,38 @@ async def async_setup(hass: HomeAssistant, config: dict) -> bool:
     """Set up the dhcp component."""
 
     async def _initialize(_):
-        dhcp_watcher = DHCPWatcher(hass, await async_get_dhcp(hass))
-        dhcp_watcher.start()
+        address_data = {}
+        integration_matchers = await async_get_dhcp(hass)
+        watchers = []
+
+        for cls in (DHCPWatcher, DeviceTrackerWatcher):
+            watcher = cls(hass, address_data, integration_matchers)
+            watcher.async_start()
+            watchers.append(watcher)
 
         def _stop(*_):
-            dhcp_watcher.stop()
-            dhcp_watcher.join()
+            for watcher in watchers:
+                watcher.async_stop()
```

Comment:
We're in sync context. We should make the context async if we need to call an async api.

Reply:
Good catch.  Will make a `stop` and `async_stop` and call the right way based on which one exists.